src/components/SidePanel.tsx

```diff
@@ -0,0 +1,138 @@
+import { FC, PropsWithChildren, ReactNode } from "react";
+import Loader from "components/Loader";
+import classnames from "classnames";
+import { Spinner } from "@canonical/react-components";
+
+interface CommonProps {
+  className?: string;
+}
+
+// Header components
+const HeaderControls: FC<PropsWithChildren & CommonProps> = ({
+  children,
+  className,
+}) => {
+  return (
+    <div className={classnames("p-panel__controls", className)}>{children}</div>
+  );
+};
+
+const HeaderTitle: FC<PropsWithChildren & CommonProps> = ({
+  children,
+  className,
+}) => {
+  return (
+    <h2 className={classnames("p-panel__title", className)}>{children}</h2>
+  );
+};
+
+const Sticky: FC<PropsWithChildren & CommonProps> = ({
+  children,
+  className,
+}) => {
+  return (
+    <div className={classnames("sticky-wrapper", className)}>{children}</div>
+  );
+};
+
+const Header: FC<PropsWithChildren & CommonProps> = ({
+  children,
+  className,
+}) => {
+  return (
+    <div className={classnames("p-panel__header", className)}>{children}</div>
+  );
+};
+
+// Panel content components
+const Container: FC<PropsWithChildren & CommonProps> = ({
+  children,
+  className,
+}) => {
+  return <div className={classnames("p-panel", className)}>{children}</div>;
+};
+
+const Content: FC<PropsWithChildren & CommonProps> = ({
+  children,
+  className,
+}) => {
+  return (
+    <div className={classnames("p-panel__content", className)}>{children}</div>
+  );
+};
+
+// Footer components
+const Footer: FC<PropsWithChildren & CommonProps> = ({
+  children,
+  className,
+}) => {
+  return (
+    <div className={classnames("panel-footer", className)}>
+      <hr />
+      {children}
+    </div>
+  );
+};
+
+interface SidePanelProps {
+  isOverlay?: boolean;
+  isSplit?: boolean;
+  children: ReactNode;
+  loading: boolean;
+  hasError: boolean;
+  className?: string;
+}
+
+const SidePanelComponent: FC<SidePanelProps> = ({
+  children,
+  isOverlay,
+  isSplit = false,
+  loading = false,
+  hasError,
+  className,
+}) => {
+  return (
+    <aside
+      className={classnames("l-aside", className, {
+        "is-narrow": !isOverlay,
+        "is-pinned": !isOverlay,
+        "is-split": isSplit,
+        "overlay-panel": isOverlay,
```

Comment:
```suggestion
        "is-overlay": isOverlay,
```
Make it consistent?

Reply:
good idea, I think the scss file name we can still keep the same as `_overlay_panel.scss`?

Reply:
Yeah, sounds good.

Reply:
Or maybe it should match the component and be called side_panel.scss?